Purchases/Public/RCPurchases.m

```diff
@@ -421,18 +421,20 @@ - (void)resetWithCompletionBlock:(nullable RCReceivePurchaserInfoBlock)completio
 }
 
 - (void)purchaserInfoWithCompletionBlock:(RCReceivePurchaserInfoBlock)completion {
-    RCPurchaserInfo *infoFromCache = [self readPurchaserInfoFromCache];
-    if (infoFromCache) {
-        RCDebugLog(@"Vending purchaserInfo from cache");
-        CALL_IF_SET_ON_MAIN_THREAD(completion, infoFromCache, nil);
-        if ([self.deviceCache isPurchaserInfoCacheStale]) {
-            RCDebugLog(@"Cache is stale, updating caches");
-            [self fetchAndCachePurchaserInfoWithCompletion:nil];
+    [self.systemInfo isApplicationBackgroundedWithCompletion:^(BOOL isAppBackgrounded) {
+        RCPurchaserInfo *infoFromCache = [self readPurchaserInfoFromCache];
+        if (infoFromCache) {
+            RCDebugLog(@"Vending purchaserInfo from cache");
+            CALL_IF_SET_ON_MAIN_THREAD(completion, infoFromCache, nil);
+            if ([self.deviceCache isPurchaserInfoCacheStaleForAppUserID:self.appUserID isAppBackgrounded:isAppBackgrounded]) {
+                RCDebugLog(@"Cache is stale, updating caches");
+                [self fetchAndCachePurchaserInfoWithCompletion:nil isAppBackgrounded:isAppBackgrounded];
+            }
+        } else {
+            RCDebugLog(@"No cached purchaser info, fetching");
+            [self fetchAndCachePurchaserInfoWithCompletion:completion isAppBackgrounded:isAppBackgrounded];
```

Comment:
really don't love making these even more complex, but I couldn't think of a way around it given that the isAppBackgrounded method is async.